opensearchpy/metrics.py

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+
+import time
+
+from events import Events
+
+
+class TimeMetrics:
```

Comment:
Create an interface and a class that implements it.

https://www.geeksforgeeks.org/abstract-classes-in-python/

```py
class Metrics(ABC)
   @abstractmethod
   def request_start()
```

```py
class MetricsEvents(Metrics):
   def request_start(self):
      self.start_time ...
```